packages/mobile/src/harmony-native/components/TextLink/TextLink.tsx

```diff
@@ -63,133 +68,89 @@ const TextLinkAnimation = (props: TextLinkAnimationProps) => {
   }))
 
   // Need to nest the AnimatedText inside a Text so the handlers & animation work
-  // while still supporting proper Text layout
+  // while still supporting proper Text layout. All this nesting is necessary
   return (
-    <Text suppressHighlighting variant={textVariant} {...other}>
-      <AnimatedText
-        style={[
-          style,
-          animatedStyles,
-          { textDecorationLine: showUnderline ? 'underline' : 'none' }
-        ]}
-      >
-        {children}
-      </AnimatedText>
+    <Text
+      suppressHighlighting
+      variant={textVariant}
+      onPressIn={(e) => {
+        animatedPressed.value = withTiming(1, motion.press)
+      }}
+      onPressOut={() => {
+        animatedPressed.value = withTiming(0, motion.press)
+      }}
+      {...other}
+    >
+      <Text>
```

Comment:
wow interesting, so why is the outer text needed?

Reply:
Without the extra Text wrapper the color animation doesn't apply to UserLinks that aren't nested inside a Text component. I'm actually not sure why that's the case